ipod-shuffle-4g.py

```diff
@@ -666,7 +725,7 @@ def check_unicode(path):
     ret_flag = False # True if there is a recognizable file within this level
     for item in os.listdir(path):
         if os.path.isfile(os.path.join(path, item)):
-            if os.path.splitext(item)[1].lower() in audio_ext+list_ext:
+            if os.path.splitext(item)[1].lower() in all_ext:
```

Comment:
I think `all_ext` is really bad to understand. Since it is only used once (i think) we do not need it and can use the better understandable name.

Why does it even check the extension here? The name of the function only checks unicode, so is it even placed correct here (in the first place, was not your initial code).

Reply:
I figured it made sense to give that union a name and define it with the other lists.

I assume it's doing the extension check to avoid renaming any non-audio files. I'll note that while the code renames non-unicode files to avoid characters that can't be handled by the iPod Shuffle file system, there are some ascii characters that can't be handled as well. Specifically, any file with these characters can't be played: `"*/:<>?|\`. That might be useful to check for as well, but I figure those make sense in a different PR.